homeassistant/components/srp_energy/sensor.py

```diff
@@ -70,98 +81,86 @@ async def async_update_data():
     )
 
     # Fetch initial data so we have data when entities subscribe
-    await coordinator.async_refresh()
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
Why do we create and update the coordinator in the sensor platform instead of when setting up the config entry in the init module?